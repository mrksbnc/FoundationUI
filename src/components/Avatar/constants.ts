export enum AvatarType {
	Image = 'image',
	Initial = 'initial',
}

export enum AvatarVariant {
	Rounded = 'rounded',
	Square = 'square',
	Flat = 'flat',
}

export enum AvatarSize {
	XS = 'xs',
	SM = 'sm',
	MD = 'md',
	LG = 'lg',
	XL = 'xl',
}

export enum AvatarIndicatorPosition {
	Top = 'top',
	Bottom = 'bottom',
}
