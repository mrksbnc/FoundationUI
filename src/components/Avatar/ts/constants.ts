export enum AvatarType {
	Image = 'image',
	Initial = 'initial',
}

export enum AvatarVariant {
	Rounded = 'rounded',
	Base = 'base',
	Flat = 'flat',
}

export enum AvatarSize {
	XS = 'xs',
	SM = 'sm',
	MD = 'md',
	LG = 'lg',
	XL = 'xl',
	XXL = 'xxl',
	XXXL = 'xxxl',
}

export enum AvatarIndicatorPosition {
	Top = 'top',
	Bottom = 'bottom',
}
