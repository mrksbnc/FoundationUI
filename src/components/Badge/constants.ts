export enum BadgeVariant {
	Default = 'default',
	Info = 'info',
	Tertiary = 'tertiary',
	Success = 'success',
	Warning = 'warning',
	Danger = 'danger',
}

export enum BadgeForm {
	Flat = 'flat',
	Base = 'base',
	Pill = 'pill',
	Circle = 'circle',
}

export enum BadgeType {
	Filled = 'filled',
	Outline = 'outline',
}

export enum BadgeSize {
	XXS = 'xxs',
	XS = 'xs',
	SM = 'sm',
	MD = 'md',
	LG = 'lg',
	XL = 'xl',
	XXL = 'xxl',
	XXXL = 'xxxl',
}
