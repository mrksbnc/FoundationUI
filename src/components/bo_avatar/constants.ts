export enum BoAvatarSize {
	extra_small = 'extra-small',
	small = 'small',
	default = 'default',
	large = 'large',
	extra_large = 'extra-large',
}

export enum BoAvatarShape {
	circle = 'circle',
	square = 'square',
	rounded = 'rounded',
}

export enum BoAvatarVariant {
	primary = 'primary',
	secondary = 'secondary',
	danger = 'danger',
	warning = 'warning',
	success = 'success',
	dark = 'dark',
}

export enum BoAvatarType {
	image = 'image',
	initials = 'initials',
	unknown = 'unknown',
}
