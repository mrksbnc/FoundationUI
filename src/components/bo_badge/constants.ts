export enum BoBadgeVariant {
	primary = 'primary',
	secondary = 'secondary',
	danger = 'danger',
	warning = 'warning',
	success = 'success',
	dark = 'dark',
}

export enum BoBadgeType {
	default = 'default',
	outline = 'outline',
}

export enum BoBadgeShape {
	pill = 'pill',
	default = 'default',
	flat = 'flat',
	circle = 'circle',
}
