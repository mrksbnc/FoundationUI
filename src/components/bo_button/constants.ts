export enum BoButtonVariant {
	primary = 'primary',
	secondary = 'secondary',
	danger = 'danger',
	warning = 'warning',
	success = 'success',
	dark = 'dark',
	link = 'link',
	link_secondary = 'link_secondary',
	link_danger = 'link_danger',
	link_warning = 'link_warning',
	link_success = 'link_success',
	link_dark = 'link_dark',
}

export enum BoButtonShape {
	default = 'default',
	pill = 'pill',
	outline = 'outline',
}
