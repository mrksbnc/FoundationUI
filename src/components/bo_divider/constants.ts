export enum BoDividerVariant {
	primary = 'primary',
	secondary = 'secondary',
	danger = 'danger',
	warning = 'warning',
	success = 'success',
	dark = 'dark',
}

export enum BoDividerType {
	default = 'default',
	dotted = 'dotted',
	dashed = 'dashed',
}
