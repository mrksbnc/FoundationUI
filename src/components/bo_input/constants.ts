export enum BoInputSize {
	small = 'small',
	default = 'default',
	large = 'large',
}

export enum BoInputState {
	none = 'none',
	valid = 'valid',
	invalid = 'invalid',
}
