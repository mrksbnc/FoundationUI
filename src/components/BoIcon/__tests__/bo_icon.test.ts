import { BoSize } from '@/constants';
import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, test } from 'vitest';
import { BoIcon, Icon } from '..';

let wrapper: ReturnType<typeof mount>;

beforeEach(() => {
	wrapper = mount(BoIcon, {
		props: {
			icon: Icon.activity,
			size: BoSize.default,
			color: '#000000',
		},
	});
});

describe('BoIcon', () => {
	test('should match snapshot', () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	test('should render the icon', () => {
		expect(wrapper.find('activity')).toBeTruthy();
	});
});
