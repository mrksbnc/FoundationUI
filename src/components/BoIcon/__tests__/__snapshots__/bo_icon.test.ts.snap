// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BoIcon > should match snapshot 1`] = `"<div style="color: rgb(0, 0, 0);" class="bo-icon size-4" aria-label="icon" type="image/svg+xml"></div>"`;
